### ISSUE TITLE
Incarnation_And_Control_Tests sometimes fail

### ISSUE THREAD

**Issue by DonOregano** [SafirSDK/safir-sdk-core#300]
Every now and then the Incarnation_And_Control_Tests unit test fails on our Raspberry Pi build server.

````
Starting nodes 0,1,2
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Checking that Server_0 has connection to ['Server_0', 'Server_1', 'Server_2']
Server_0 has connection to (only) ['Server_0', 'Server_1', 'Server_2']
Stopping node 2 from System_0
Trying to stop node 4848746291424386580 from instance 0
Checking that Server_0 has connection to ['Server_0', 'Server_1']
Server_0 has connection to (only) ['Server_0', 'Server_1']
Starting node 2
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Checking that Server_0 has connection to ['Server_0', 'Server_1', 'Server_2']
Server_0 has connection to (only) ['Server_0', 'Server_1', 'Server_2']
Stopping the system from Server_0, Server_0 should not go down due to being started with --ignore-control-cmd
Trying to stop the system from instance 0
Checking that Server_0 has only to itself
Server_0 has connection to (only) ['Server_0']
Starting Server_1 and Server_2
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Checking that Server_0 has only to itself
Server_0 has connection to (only) ['Server_0']
Checking that Server_1 has connection to ['Server_1', 'Server_2']
Server_1 has connection to (only) ['Server_1', 'Server_2']
Hardkill Server_0
Terminating all processes
 Terminating safir_control
   Terminate successful
Starting Server_0
Launching safir_control
Launching dope_main
Waiting for safir_control to be ready
 dose_main seems to be ready
Error: Server_0 does not have contact with (only) ['Server_0', 'Server_1', 'Server_2'] we have connection to: ['Server_0', 'Server_1']
Terminating all processes
 Terminating safir_control
   Terminate successful
Terminating all processes
 Terminating safir_control
   Terminate successful
Terminating all processes
 Terminating safir_control
   Terminate successful
````

**Comment by DonOregano** [comment]
This happened on an ubuntu 14.04 build server also. It's not unlikely to have something to do with the fix for #275. Needs investigation!

**Comment by DonOregano** [comment]
When debugging this it has proven to be two different issues. One where System Picture could get confused when there are two nodes starting at the same time and a third one that is blacklisted. In short one of the new nodes would think that it was elected and the other would think that the blacklisted one was elected. This issue is resolved by f179cb8c543e5884a982beae064be3c526cd43b1.

The other issue was that communication would sometimes fail to bind the socket (an exception would be thrown), which caused dose_main to crash. This issue is resolved by db5a8fbb547d0141560c715f8ff214ed64b04566.
